client/src/features/forgotpassword/ForgotPasswordReset.tsx

```diff
@@ -1,101 +1,177 @@
-import React, { useState } from 'react';
-import { Link } from 'react-router-dom';
+import React, { useState, useEffect } from 'react';
+import { Link, useParams } from 'react-router-dom';
 import LeftPart from '../../common/components/LeftPart';
 import IPasswords from '../../interfaces/passwords';
 import { FaUnlockAlt } from 'react-icons/fa';
 import { AiFillEye, AiFillEyeInvisible } from 'react-icons/ai';
+import ForgotPasswordInvalidUrl from './ForgotPasswordInvalidUrl';
+import axios from 'axios';
+const PASSWORD_REGEX =
+  /^(?=.*[a-z])(?=.*[0-9])[a-zA-Z0-9~`! @#\$%\^&*()_\-\+=\{\[\}\]\|\\:;"'<,>\.\?/]{6,32}$/;
```

Comment:
@miyabitanimchi You could have used the `validator` npm package here as well, but this is also fine👍